### PR TITLE
Lifecycle test changes for openebs lvm plugin

### LINKED ISSUE #398: Complete lifecycle of CSI plugNplay with anyone csi driver
**Feature Description:**
Complete lifecycle of CSI plugNplay with some CSI driver where all the features e2e can be checked.

**Why this feature is needed(give scenarios or use cases):**

This is required to check the feasibility of CSI plugNplay in production env

### PR DESCRIPTION
**Pull request by asifdxtreme**
**What type of PR is this?**
 /kind enhancement

**What this PR does / why we need it**:
This PR adds a new version of soda-csi-provisioner which is based on a the v2.1 version of https://github.com/kubernetes-csi/external-provisioner.
To test the soda csi-pnp with openebs lvm plugin, a new version of csi-provisoner was required.
Hence this the updated base code of v2.1 with soda changes.

**Which issue(s) this PR fixes**:

Fixes #398 

**Test Report Added?**:
/kind TESTED

**Test Report**:
Test steps and verification results can be found here  https://github.com/sodafoundation/nbp/issues/398#issuecomment-786156132
**Special notes for your reviewer**:
We need to find the best way to manage the base code of several version of these side cars, till the resolution we need to manage the code like this.

### CODE REVIEW

**Comment by asifdxtreme** [comment]
Actual changes are in these two files

 - https://github.com/sodafoundation/nbp/pull/404/files#diff-646648329cbf7141ded1d0ab07dd9040147675d92c5b454c8ec96fe461c348ba
 - https://github.com/sodafoundation/nbp/pull/404/files#diff-632129a04600c37536a6d36c8471aeb2cecd08fb3d4a0d7012f8b70d449ae5eb

File: csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go [hunk 1]
```diff
@@ -494,13 +499,56 @@ type prepareProvisionResult struct {
...
 	if sc == nil {
 		return nil, controller.ProvisioningFinished, errors.New("storage class was nil")
 	}
 
+	//SODA-CHANGE
+	backendDriverName, err := GetDriverName(p.grpcClient, operationTimeout)
+	if err != nil {
+		klog.Fatalf("Error getting CSI driver name: %s", err)
+	}
+	klog.Infof("The Backend Driver Name and provisioner name is : %s , %s", backendDriverName, p.driverName)
```
Comment (sushanthakumar) [inline csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go:525]: p.driverName  is soda-csi?

File: csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go [hunk 2]
```diff
@@ -494,13 +499,56 @@ type prepareProvisionResult struct {
...
+
+	if sc.Provisioner == "soda-csi" {
+		for k, v := range sc.Parameters {
+			klog.Infof("The parameters in the StorageClass are  : %s ===== %s", k, v)
+			if k == "profile" {
+
+				response, err := http.Get("http://soda-proxy:50029/getprofile/" + v)
```
Comment (sushanthakumar) [inline csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go]: Suggest to define proxy port constant for 50029

File: csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go [hunk 3]
```diff
@@ -494,13 +499,56 @@ type prepareProvisionResult struct {
...
+				if err != nil {
+					klog.Infof("Got error in GetProfile  : %s ===== %s", err.Error())
+				} else {
+					data, _ := ioutil.ReadAll(response.Body)
+					var customProperties *CustomPropertiesSpec
+					json.Unmarshal(data, &customProperties)
+					klog.Infof("The profile name received in the storageClass is: %s", customProperties.GetDriverPreference())
```
Comment (sushanthakumar) [inline csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go]: profile name or driver name?

File: csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go [hunk 4]
```diff
@@ -494,13 +499,56 @@ type prepareProvisionResult struct {
...
+					if backendDriverName != customProperties.GetDriverPreference() {
```
Comment (sushanthakumar) [inline csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go]: multiple calls to customProperties.GetDriverPreference() can be avoided for line 539 and 540

**Comment by sushanthakumar** [comment]
Pls raise another PR to add updated yaml files for the deployment of  all tested csi drivers which can be directly used by anyone

**Comment by asifdxtreme** [comment]
> Pls raise another PR to add updated yaml files for the deployment of all tested csi drivers which can be directly used by anyone

A follow up Issue has been raised to track this https://github.com/sodafoundation/nbp/issues/406

**Comment by asifdxtreme** [comment]
@sushanthakumar please review the changes, I have added a  new commit with just soda-changes and also optimized the flow for performance and maintainability.

**Comment by sushanthakumar** [comment]
LGTM

**Review by kumarashit — APPROVED** [review]
LGTM

**Pull request merged**